src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs

```diff
@@ -120,7 +120,14 @@ public void Add(MessageEndpointMapping mapping)
         /// </summary>
         protected override void BaseAdd(ConfigurationElement element)
         {
-            BaseAdd(element, true);
+            try
+            {
+                BaseAdd(element, true);
+            }
+            catch (ConfigurationErrorsException e)
+            {
+                throw new Exception($"An ambigious message endpoint mapping has been defined at line: {e.Line}. Check the 'MessageEndpointMappings' section in {e.Filename} for conflicting mappings.", e);
```

Comment:
I couldn't find more documentation on this and what kind of errors could cause an exception. Already existing keys seem to be the only "invalid" input mentioned in the docs on the `BaseAdd` method.